x-pack/libbeat/management/managerV2.go

```diff
@@ -23,6 +27,13 @@ import (
 	"github.com/elastic/elastic-agent-libs/logp"
 )
 
+// unitKey is used to identify a unique unit in a map
+// the `ID` of a unit in itself is not unique without its type, only `Type` + `ID` is unique
+type unitKey struct {
+	Type client.UnitType
+	ID   string
```

Comment:
Is there anything on the agent side enforcing that this tuple is unique per unit?

Trying to trace where this is coming from and I see this on the agent side: https://github.com/elastic/elastic-agent/blob/7e376c3eec28d0bbd9064930042a2225e9a29dbc/pkg/component/component.go#L235-L241

```go
				units = append(units, Unit{
					ID:       fmt.Sprintf("%s-%s-%s", inputType, outputName, input.id),
					Type:     client.UnitTypeInput,
					LogLevel: input.logLevel,
					Config:   cfg,
					Err:      cfgErr,
				})
```

The uniqueness is coming from the `input.id` primarily, which I see parsed out of the agent policy with a uniqueness check so it seems to be there: https://github.com/elastic/elastic-agent/blob/7e376c3eec28d0bbd9064930042a2225e9a29dbc/pkg/component/component.go#L511-L522

```go
		idRaw, ok := input[idKey]
		if !ok {
			// no ID; fallback to type
			idRaw = t
		}
		id, ok := idRaw.(string)
		if !ok {
			return nil, fmt.Errorf("invalid 'inputs.%d.id', expected a string not a %T", idx, idRaw)
		}
		if hasDuplicate(outputsMap, id) {
			return nil, fmt.Errorf("invalid 'inputs.%d.id', has a duplicate id %q (id is required to be unique)", idx, id)
		}
```

Was this uniqueness check there before? Does it work with standalone agent policies that omit the ID field? We don't seem to document the need for it in our standalone agent docs so likely we will encounter agent policies with missing input IDs in the wild, possibly lots of them.

Reply:
ID is a required value on standalone agents now, so they will error and then they will need to modify the `elastic-agent.yml` to set an `id` on the input definition.

The Elastic Agent does enforce that a Type/ID is unique.